models/models.go

```diff
@@ -783,55 +792,83 @@ func (m *RouteManager) solveChallenges(clients map[acme.Challenge]*acme.Client,
 func (m *RouteManager) deployCertificate(route Route, cert acme.CertificateResource) error {
 	lsession := m.logger.Session("deploy-certificate", lager.Data{
 		"instance-id": route.InstanceId,
+		"domains":     route.GetDomains(),
+		"dist":        route.DistId,
 	})
 
+	lsession.Info("get-cert-expiry")
 	expires, err := acme.GetPEMCertExpiration(cert.Certificate)
 	if err != nil {
-		lsession.Error("get-cert-expiry", err)
+		lsession.Error("get-cert-expiry-err", lager.Data{
+			"err": err,
+		})
 		return err
 	}
 
 	name := fmt.Sprintf("cdn-route-%s-%s", route.InstanceId, expires.Format("2006-01-02_15-04-05"))
-
-	m.logger.Info("Uploading certificate to IAM", lager.Data{"name": name})
-
+	lsession.Info("iam-upload-certificate", lager.Data{"name": name})
 	certId, err := m.iam.UploadCertificate(name, cert)
 	if err != nil {
-		lsession.Error("iam-upload-certificate", err)
+		lsession.Error("iam-upload-certificate-err", lager.Data{
+			"name": name,
+			"err":  err,
+		})
 		return err
 	}
 
-	return m.cloudFront.SetCertificateAndCname(route.DistId, certId, route.GetDomains())
+	lsession.Info("iam-set-certificate-and-cname", lager.Data{
+		"name":    name,
+		"cert_id": certId,
+	})
+	err = m.cloudFront.SetCertificateAndCname(route.DistId, certId, route.GetDomains())
+	if err != nil {
+		lsession.Info("iam-set-certificate-and-cname-err", lager.Data{
+			"name":    name,
+			"cert_id": certId,
+			"err":     err,
+		})
+		return err
+	}
+	return nil
 }
 
 func (m *RouteManager) ensureChallenges(route *Route, client *acme.Client, update bool) error {
 	lsession := m.logger.Session("ensure-challenges", lager.Data{
 		"instance-id": route.InstanceId,
```

Comment:
Could we include the domain here too? If we include it everywhere (ideally with the same key, if it makes sense) we can easily grep for it

Reply:
Added